app/api/mail.py

```diff
@@ -0,0 +1,41 @@
+from fastapi import APIRouter, Depends, HTTPException, Response
+from ..auth_helpers import get_google_credentials
+from ..auth_helpers import authorize_admin
+from ..models import MailPayload, AccessTokenPayload
+
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+from email.message import EmailMessage
+import base64
+
+router = APIRouter()
+
+@router.post('/send-mail/')
+def send_mail(payload: MailPayload, token: AccessTokenPayload = Depends(authorize_admin)):
+    try:
```

Comment:
There is a clean way, I separated them because they check to completely different things. However, on second thought the first check would possibly be better of in inside the ```get_google_credentials``` function :thinking:  But I guess it is a bit up to how we want to do stuff like this :shrug:

Reply:
I understand! I think I would prefer the try/except inside the router like it is now over moving it inside `get_google_credentials` though, since that would be raising an HTTP-error inside a function that is not a router. And I think it's maybe best to separate those in case we want the credentials outside of a router later for example.

So maybe it's just best to leave it as is now, since it checks for two different exceptions and I if I remember correctly, I _think_ I have heard/read once that it's bad practice to just catch all errors and not explicitly declare what errors that happen (not sure about that though🤷‍♂️)

Reply:
Again a bit quick on the trigger, I moved it into ```get_google_credentials```. I do agree though. One can do 
```
try:
    #stuff
except error1:
    #Handle error 1
except error2:
    #Handle error 2
```
However, I think this is better on code that does one thing but can result in different errors. In this scenario the code does to different things, therefore I think it was best the way it was. 

I think I just managed to agree on myself here :clap:

Reply:
Okay, yeah I agree🙌